portafly/src/components/data-list/reducers/TableReducer.ts

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import {
+  ActionHandlers,
+  Action,
+  createReducer
+} from 'utils'
+
+export type TableState = {
+  initialRows: Readonly<any[]>
+  columns: any[],
+  rows: any[],
+}
+
+// Action Handlers
+const SELECT_ONE = 'SELECT_ONE'
+const SELECT_PAGE = 'SELECT_PAGE'
+const SELECT_ALL = 'SELECT_ALL'
+
+const tableActionHandlers: ActionHandlers<TableState, any> = {
+  [SELECT_ONE]: (state, action) => {
+    const { id, selected } = action.payload
+    const newRows = state.rows.map((r) => (r.id === id ? { ...r, selected } : r))
+    return { ...state, rows: newRows }
+  },
+  [SELECT_PAGE]: (state, action) => {
+    // TODO: keep track of visibleRows in Context?
+    const visibleRows = action.payload as any[]
+    const newRows = state.initialRows.map((r) => {
+      const isRowVisible = visibleRows.find((vR) => vR.id === r.id)
+      return { ...r, selected: isRowVisible }
+    })
+    // OR
+    // const newRows = state.initialRows.map((r) => ({
+    //   ...r,
+    //   selected: visibleRows.find((vR) => vR.id === r.id)
+    // }))
```

Comment:
I personally fancy more this option

Reply:
My subconscious probably left it for you 😂